src/main/java/com/bakdata/streams_store/InMemoryCache.java

```diff
@@ -0,0 +1,59 @@
+package com.bakdata.streams_store;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.lang.ref.SoftReference;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryCache implements Cache {
```

Comment:
Looks like you're using some sort of In memory cache? Does this mean if the pod restarts, you would lose all the cached data?

Reply:
Yes.  The data is cached in case the consumer goes down, in which case the app will continue serving the cached data.  If the pod restarts, the app will reconnect with the consumer and get the latest data streams.

Reply:
>  If the pod restarts, the app will reconnect with the consumer and get the latest data streams.

Right, but what if the latest data stream is null? You should still have the latest valid data. If this is saved in memory, you would lose that.

Reply:
I think it would be great to utilize the REDIS server that's available for caching these values.

Reply:
Yeah that would be a cool feature.  I'll work on that in a separate PR.  At least for now the app is a lot more resilient than it was.